src/lib/app-provider/hooks/useChainRecordAsset.ts

```diff
@@ -0,0 +1,36 @@
+import type { Asset } from "@chain-registry/types";
+import { useWallet } from "@cosmos-kit/react";
+import { useCallback } from "react";
+
+import type { SnakeToCamelCaseNested } from "lib/types";
+import { snakeToCamel } from "lib/utils";
+
+interface ChainRecordAsset extends Asset {
+  precision: number;
+}
+
+type InternalChainRecordAsset = SnakeToCamelCaseNested<ChainRecordAsset>;
+
+export const useChainRecordAsset = () => {
+  const { currentChainRecord } = useWallet();
+
+  return useCallback(
+    (denom: string): InternalChainRecordAsset | null => {
+      const assetInfo = currentChainRecord?.assetList.assets.find(
+        (asset) => asset.base === denom
+      );
+      return assetInfo
+        ? (snakeToCamel({
+            ...assetInfo,
+            precision:
+              assetInfo?.denom_units.find(
+                (unit) =>
+                  unit.denom ===
+                  (denom[0] === "u" ? denom.slice(1) : "u".concat(denom))
```

Comment:
I think "u".concat(denom) is quite strange. maybe just return only denom instead or add todo for revisiting

Reply:
the logic is a bit weird here -- if "denom" starts with "u", it is sliced out. But if "denom" doesn't start with "u", "u" is appended to the front of the denom

Reply:
in chain record the exponent field is inverse of precision 
e.g. (Precision) uToken = 6 token = 0,  (Exponent) uToken = 0 token = 6

Reply:
Understand what you mean, just worry about the case where there is no uToken and token or other edge case